etree.go

```diff
@@ -894,6 +921,58 @@ func (c *CharData) writeTo(w *bufio.Writer, s *WriteSettings) {
 	escapeString(w, c.Data, m)
 }
 
+// NewPlainTextData creates a parentless XML character data entity.
+func NewPlainTextData(data string) *PlainTextData {
+	return newPlainTextData(data, false, nil)
+}
+
+// newPlainTextData creates an XML character data entity and binds it to a parent
+// element. If parent is nil, the PlainTextData token remains unbound.
+func newPlainTextData(data string, whitespace bool, parent *Element) *PlainTextData {
+	c := &PlainTextData{
+		Data:       data,
+		whitespace: whitespace,
+		parent:     parent,
+	}
+	if parent != nil {
+		parent.addChild(c)
+	}
+	return c
+}
+
+// CreatePlainTextData creates an XML character data entity and adds it as a child
+// of element e.
+func (e *Element) CreatePlainTextData(data string) *PlainTextData {
+	return newPlainTextData(data, false, e)
+}
+
+// dup duplicates the character data.
+func (c *PlainTextData) dup(parent *Element) Token {
+	return &PlainTextData{
+		Data:       c.Data,
+		whitespace: c.whitespace,
+		parent:     parent,
+	}
+}
+
+// Parent returns the character data token's parent element, or nil if it has
+// no parent.
+func (c *PlainTextData) Parent() *Element {
+	return c.parent
+}
+
+// setParent replaces the character data token's parent.
+func (c *PlainTextData) setParent(parent *Element) {
+	c.parent = parent
+}
+
+// writeTo serializes the character data entity to the writer.
+func (c *PlainTextData) writeTo(w *bufio.Writer, s *WriteSettings) {
+	w.Write(cdataStart)
+	w.WriteString(c.Data)
+	w.Write(cdataEnd)
+}
+
```

Comment:
No longer necessary.  But you will need to modify the `CharData` methods to do different things depending on whether `plain` is true.